### ISSUE TITLE
Issue with blank lines before a function (flake8)

### ISSUE THREAD

**Issue by amyheather** [lintquarto/lintquarto#14]
**Issue identified for flake8**

Example:

`````
---
title: "Example"
---

This file contains some issues for the linter to identify.

```{python}
very_long_line = "This long string exceeds the maximum allowed characters per line."
```

There's some more issues below...

```{python}
def add_numbers(a, b):
    return a + b

add_numbers(3, 5)
```
`````

Run:

```
lintquarto flake8 file.qmd
```

Output:

```
examples/flake8_example.qmd:8:80: E501 line too long (84 > 79 characters)
examples/flake8_example.qmd:14:1: E302 expected 2 blank lines, found 0
examples/flake8_example.qmd:17:1: E305 expected 2 blank lines after class or function definition, found 1
```

Issue:

There should not be an error for expecting 2 blank lines before the function, as it is in a new cell. If you put the function into a blank python file, it wouldn't petition about needing lines before, unless you put any comments above it.

**Comment by amyheather** [comment]
**Universally disable E302**: Not a fan of this option, as it means valid E302 issues don't get flagged.

**Add `# noqa: E302` to the end of each first line**: Simple and quick, but would show up in the linting messages.

- Although, could append something before or after that is like "REMOVEME" or something, so it scrapes it from the printed output
- Although, if someone had that in their code already, it would alter their code in the output.

**Filter E302 messages at python cell starts from the linter output:** Cleanest option, though complex and takes some time.

**Comment by amyheather** [comment]
Addressed issue, not removing the added flags for now, although could consider doing so later if think its an issue.
